frb_codegen/src/generator/dart/mod.rs

```diff
@@ -467,53 +620,106 @@ fn generate_api_func(func: &IrFunc, ir_file: &IrFile) -> GeneratedApiFunc {
     }
 }
 
-fn generate_api2wire_func(ty: &IrType, ir_file: &IrFile, block_index: BlockIndex) -> String {
-    if let Some(body) = TypeDartGenerator::new(ty.clone(), ir_file, None).api2wire_body(block_index)
-    {
+fn rust_wire_to_dart_wire(ty: &str) -> Cow<str> {
+    let ty = ty.trim();
+    if matches!(ty, "void" | "WireSyncReturnStruct") {
+        return ty.into();
+    }
+    if ty.starts_with("*mut") {
+        return format!("int /* {} */", ty).into();
+    }
+    return format!("dynamic /* {} */", ty).into();
+}
+
+fn generate_wasm_wire_func_decl(func: &IrFuncLike) -> String {
+    format!(
+        "external {} {name}({});",
+        rust_wire_to_dart_wire(&func.output),
+        func.inputs
+            .iter()
+            .map(|(key, ty)| format!("{} {}", ty, key))
+            .join(","),
+        name = func.name,
+    )
+}
+
+fn generate_wasm_wire_func_method(func: &IrFuncLike) -> String {
```

Comment:
In this file (mod.rs) (and probably other files two), we have seen wasm things mixed with io things. That may lead to bad code smell in the future... So maybe we can avoid it: For example:

Can we extract all wasm things into wasm.rs (just an naive example of naming), io things to io.rs, common things to common.rs?

Reply:
I have refactored the most offending items, but there may be a need for a general cleanup PR later.